### PR TITLE
회원가입 기능 추가 및 의존성 추가

### PR DESCRIPTION
**Pull request by dlrtn**
회원가입 기능 추가하였으나 404 에러 발생합니다. 버그 찾는 중이나 찾지 못했습니다. 내일 오전 중으로 디버깅 작업 예정입니다. 아래는 테스트/디버깅 진행 결과인데, 참고하시어 문제되는 부분 있으면 말씀해주시면 좋겠습니다.

### CODE REVIEW

**Comment by dlrtn** [comment]
디버깅 결과

01:01:16.013 [Test worker] DEBUG org.springframework.test.context.BootstrapUtils - Instantiating CacheAwareContextLoaderDelegate from class [org.springframework.test.context.cache.DefaultCacheAwareContextLoaderDelegate]
01:01:16.034 [Test worker] DEBUG org.springframework.test.context.BootstrapUtils - Instantiating BootstrapContext using constructor [public org.springframework.test.context.support.DefaultBootstrapContext(java.lang.Class,org.springframework.test.context.CacheAwareContextLoaderDelegate)]
01:01:16.085 [Test worker] DEBUG org.springframework.test.context.BootstrapUtils - Instantiating TestContextBootstrapper for test class [com.dlrtn.websocket.chat.WebsocketChatApplicationTests] from class [org.springframework.boot.test.context.SpringBootTestContextBootstrapper]
01:01:16.100 [Test worker] INFO org.springframework.boot.test.context.SpringBootTestContextBootstrapper - Neither @ContextConfiguration nor @ContextHierarchy found for test class [com.dlrtn.websocket.chat.WebsocketChatApplicationTests], using SpringBootContextLoader
01:01:16.106 [Test worker] DEBUG org.springframework.test.context.support.AbstractContextLoader - Did not detect default resource location for test class [com.dlrtn.websocket.chat.WebsocketChatApplicationTests]: class path resource [com/dlrtn/websocket/chat/WebsocketChatApplicationTests-context.xml] does not exist
01:01:16.107 [Test worker] DEBUG org.springframework.test.context.support.AbstractContextLoader - Did not detect default resource location for test class [com.dlrtn.websocket.chat.WebsocketChatApplicationTests]: class path resource [com/dlrtn/websocket/chat/WebsocketChatApplicationTestsContext.groovy] does not exist
01:01:16.107 [Test worker] INFO org.springframework.test.context.support.AbstractContextLoader - Could not detect default resource locations for test class [com.dlrtn.websocket.chat.WebsocketChatApplicationTests]: no resource found for suffixes {-context.xml, Context.groovy}.
01:01:16.109 [Test worker] INFO org.springframework.test.context.support.AnnotationConfigContextLoaderUtils - Could not detect default configuration classes for test class [com.dlrtn.websocket.chat.WebsocketChatApplicationTests]: WebsocketChatApplicationTests does not declare any static, non-private, non-final, nested classes annotated with @Configuration.
01:01:16.178 [Test worker] DEBUG org.springframework.test.context.support.ActiveProfilesUtils - Could not find an 'annotation declaring class' for annotation type [org.springframework.test.context.ActiveProfiles] and class [com.dlrtn.websocket.chat.WebsocketChatApplicationTests]
01:01:16.251 [Test worker] DEBUG org.springframework.context.annotation.ClassPathScanningCandidateComponentProvider - Identified candidate component class: file [C:\Users\user\Desktop\web-socket-chat\build\classes\java\main\com\dlrtn\websocket\chat\WebsocketChatApplication.class]
01:01:16.253 [Test worker] INFO org.springframework.boot.test.context.SpringBootTestContextBootstrapper - Found @SpringBootConfiguration com.dlrtn.websocket.chat.WebsocketChatApplication for test class com.dlrtn.websocket.chat.WebsocketChatApplicationTests
01:01:16.383 [Test worker] DEBUG org.springframework.boot.test.context.SpringBootTestContextBootstrapper - @TestExecutionListeners is not present for class [com.dlrtn.websocket.chat.WebsocketChatApplicationTests]: using defaults.
01:01:16.384 [Test worker] INFO org.springframework.boot.test.context.SpringBootTestContextBootstrapper - Loaded default TestExecutionListener class names from location [META-INF/spring.factories]: [org.springframework.boot.test.autoconfigure.restdocs.RestDocsTestExecutionListener, org.springframework.boot.test.autoconfigure.web.client.MockRestServiceServerResetTestExecutionListener, org.springframework.boot.test.autoconfigure.web.servlet.MockMvcPrintOnlyOnFailureTestExecutionListener, org.springframework.boot.test.autoconfigure.web.servlet.WebDriverTestExecutionListener, org.springframework.boot.test.autoconfigure.webservices.client.MockWebServiceServerTestExecutionListener, org.springframework.boot.test.mock.mockito.MockitoTestExecutionListener, org.springframework.boot.test.mock.mockito.ResetMocksTestExecutionListener, org.springframework.test.context.web.ServletTestExecutionListener, org.springframework.test.context.support.DirtiesContextBeforeModesTestExecutionListener, org.springframework.test.context.event.ApplicationEventsTestExecutionListener, org.springframework.test.context.support.DependencyInjectionTestExecutionListener, org.springframework.test.context.support.DirtiesContextTestExecutionListener, org.springframework.test.context.transaction.TransactionalTestExecutionListener, org.springframework.test.context.jdbc.SqlScriptsTestExecutionListener, org.springframework.test.context.event.EventPublishingTestExecutionListener]
01:01:16.404 [Test worker] INFO org.springframework.boot.test.context.SpringBootTestContextBootstrapper - Using TestExecutionListeners: [org.springframework.test.context.web.ServletTestExecutionListener@22db8f4, org.springframework.test.context.support.DirtiesContextBeforeModesTestExecutionListener@2b46a8c1, org.springframework.test.context.event.ApplicationEventsTestExecutionListener@1d572e62, org.springframework.boot.test.mock.mockito.MockitoTestExecutionListener@29caf222, org.springframework.boot.test.autoconfigure.SpringBootDependencyInjectionTestExecutionListener@46cf05f7, org.springframework.test.context.support.DirtiesContextTestExecutionListener@5851bd4f, org.springframework.test.context.transaction.TransactionalTestExecutionListener@7cd1ac19, org.springframework.test.context.jdbc.SqlScriptsTestExecutionListener@2f40a43, org.springframework.test.context.event.EventPublishingTestExecutionListener@3caa4757, org.springframework.boot.test.autoconfigure.restdocs.RestDocsTestExecutionListener@69c43e48, org.springframework.boot.test.autoconfigure.web.client.MockRestServiceServerResetTestExecutionListener@1804f60d, org.springframework.boot.test.autoconfigure.web.servlet.MockMvcPrintOnlyOnFailureTestExecutionListener@3a80515c, org.springframework.boot.test.autoconfigure.web.servlet.WebDriverTestExecutionListener@547e29a4, org.springframework.boot.test.autoconfigure.webservices.client.MockWebServiceServerTestExecutionListener@1c807b1d, org.springframework.boot.test.mock.mockito.ResetMocksTestExecutionListener@238b521e]
01:01:16.408 [Test worker] DEBUG org.springframework.test.context.support.AbstractDirtiesContextTestExecutionListener - Before test class: context [DefaultTestContext@6ebd78d1 testClass = WebsocketChatApplicationTests, testInstance = [null], testMethod = [null], testException = [null], mergedContextConfiguration = [WebMergedContextConfiguration@436390f4 testClass = WebsocketChatApplicationTests, locations = '{}', classes = '{class com.dlrtn.websocket.chat.WebsocketChatApplication}', contextInitializerClasses = '[]', activeProfiles = '{}', propertySourceLocations = '{}', propertySourceProperties = '{org.springframework.boot.test.context.SpringBootTestContextBootstrapper=true}', contextCustomizers = set[org.springframework.boot.test.autoconfigure.actuate.metrics.MetricsExportContextCustomizerFactory$DisableMetricExportContextCustomizer@6aecbb8d, org.springframework.boot.test.autoconfigure.properties.PropertyMappingContextCustomizer@0, org.springframework.boot.test.autoconfigure.web.servlet.WebDriverContextCustomizerFactory$Customizer@62163b39, org.springframework.boot.test.context.filter.ExcludeFilterContextCustomizer@207b8649, org.springframework.boot.test.json.DuplicateJsonObjectContextCustomizerFactory$DuplicateJsonObjectContextCustomizer@57eda880, org.springframework.boot.test.mock.mockito.MockitoContextCustomizer@0, org.springframework.boot.test.web.client.TestRestTemplateContextCustomizer@cb191ca, org.springframework.boot.test.context.SpringBootTestArgs@1, org.springframework.boot.test.context.SpringBootTestWebEnvironment@6ca18a14], resourceBasePath = 'src/main/webapp', contextLoader = 'org.springframework.boot.test.context.SpringBootContextLoader', parent = [null]], attributes = map['org.springframework.test.context.web.ServletTestExecutionListener.activateListener' -> true]], class annotated with @DirtiesContext [false] with mode [null].

**Comment by dlrtn** [comment]
테스트 결과

00:55:50.908 [Test worker] DEBUG org.springframework.test.context.BootstrapUtils - Instantiating CacheAwareContextLoaderDelegate from class [org.springframework.test.context.cache.DefaultCacheAwareContextLoaderDelegate]
00:55:50.920 [Test worker] DEBUG org.springframework.test.context.BootstrapUtils - Instantiating BootstrapContext using constructor [public org.springframework.test.context.support.DefaultBootstrapContext(java.lang.Class,org.springframework.test.context.CacheAwareContextLoaderDelegate)]
00:55:50.963 [Test worker] DEBUG org.springframework.test.context.BootstrapUtils - Instantiating TestContextBootstrapper for test class [com.dlrtn.websocket.chat.WebsocketChatApplicationTests] from class [org.springframework.boot.test.context.SpringBootTestContextBootstrapper]
00:55:50.979 [Test worker] INFO org.springframework.boot.test.context.SpringBootTestContextBootstrapper - Neither @ContextConfiguration nor @ContextHierarchy found for test class [com.dlrtn.websocket.chat.WebsocketChatApplicationTests], using SpringBootContextLoader
00:55:50.984 [Test worker] DEBUG org.springframework.test.context.support.AbstractContextLoader - Did not detect default resource location for test class [com.dlrtn.websocket.chat.WebsocketChatApplicationTests]: class path resource [com/dlrtn/websocket/chat/WebsocketChatApplicationTests-context.xml] does not exist
00:55:50.985 [Test worker] DEBUG org.springframework.test.context.support.AbstractContextLoader - Did not detect default resource location for test class [com.dlrtn.websocket.chat.WebsocketChatApplicationTests]: class path resource [com/dlrtn/websocket/chat/WebsocketChatApplicationTestsContext.groovy] does not exist
00:55:50.985 [Test worker] INFO org.springframework.test.context.support.AbstractContextLoader - Could not detect default resource locations for test class [com.dlrtn.websocket.chat.WebsocketChatApplicationTests]: no resource found for suffixes {-context.xml, Context.groovy}.
00:55:50.986 [Test worker] INFO org.springframework.test.context.support.AnnotationConfigContextLoaderUtils - Could not detect default configuration classes for test class [com.dlrtn.websocket.chat.WebsocketChatApplicationTests]: WebsocketChatApplicationTests does not declare any static, non-private, non-final, nested classes annotated with @Configuration.
00:55:51.048 [Test worker] DEBUG org.springframework.test.context.support.ActiveProfilesUtils - Could not find an 'annotation declaring class' for annotation type [org.springframework.test.context.ActiveProfiles] and class [com.dlrtn.websocket.chat.WebsocketChatApplicationTests]
00:55:51.115 [Test worker] DEBUG org.springframework.context.annotation.ClassPathScanningCandidateComponentProvider - Identified candidate component class: file [C:\Users\user\Desktop\web-socket-chat\build\classes\java\main\com\dlrtn\websocket\chat\WebsocketChatApplication.class]
00:55:51.118 [Test worker] INFO org.springframework.boot.test.context.SpringBootTestContextBootstrapper - Found @SpringBootConfiguration com.dlrtn.websocket.chat.WebsocketChatApplication for test class com.dlrtn.websocket.chat.WebsocketChatApplicationTests
00:55:51.241 [Test worker] DEBUG org.springframework.boot.test.context.SpringBootTestContextBootstrapper - @TestExecutionListeners is not present for class [com.dlrtn.websocket.chat.WebsocketChatApplicationTests]: using defaults.
00:55:51.241 [Test worker] INFO org.springframework.boot.test.context.SpringBootTestContextBootstrapper - Loaded default TestExecutionListener class names from location [META-INF/spring.factories]: [org.springframework.boot.test.autoconfigure.restdocs.RestDocsTestExecutionListener, org.springframework.boot.test.autoconfigure.web.client.MockRestServiceServerResetTestExecutionListener, org.springframework.boot.test.autoconfigure.web.servlet.MockMvcPrintOnlyOnFailureTestExecutionListener, org.springframework.boot.test.autoconfigure.web.servlet.WebDriverTestExecutionListener, org.springframework.boot.test.autoconfigure.webservices.client.MockWebServiceServerTestExecutionListener, org.springframework.boot.test.mock.mockito.MockitoTestExecutionListener, org.springframework.boot.test.mock.mockito.ResetMocksTestExecutionListener, org.springframework.test.context.web.ServletTestExecutionListener, org.springframework.test.context.support.DirtiesContextBeforeModesTestExecutionListener, org.springframework.test.context.event.ApplicationEventsTestExecutionListener, org.springframework.test.context.support.DependencyInjectionTestExecutionListener, org.springframework.test.context.support.DirtiesContextTestExecutionListener, org.springframework.test.context.transaction.TransactionalTestExecutionListener, org.springframework.test.context.jdbc.SqlScriptsTestExecutionListener, org.springframework.test.context.event.EventPublishingTestExecutionListener]
00:55:51.260 [Test worker] INFO org.springframework.boot.test.context.SpringBootTestContextBootstrapper - Using TestExecutionListeners: [org.springframework.test.context.web.ServletTestExecutionListener@46cf05f7, org.springframework.test.context.support.DirtiesContextBeforeModesTestExecutionListener@5851bd4f, org.springframework.test.context.event.ApplicationEventsTestExecutionListener@7cd1ac19, org.springframework.boot.test.mock.mockito.MockitoTestExecutionListener@2f40a43, org.springframework.boot.test.autoconfigure.SpringBootDependencyInjectionTestExecutionListener@3caa4757, org.springframework.test.context.support.DirtiesContextTestExecutionListener@69c43e48, org.springframework.test.context.transaction.TransactionalTestExecutionListener@1804f60d, org.springframework.test.context.jdbc.SqlScriptsTestExecutionListener@3a80515c, org.springframework.test.context.event.EventPublishingTestExecutionListener@547e29a4, org.springframework.boot.test.autoconfigure.restdocs.RestDocsTestExecutionListener@1c807b1d, org.springframework.boot.test.autoconfigure.web.client.MockRestServiceServerResetTestExecutionListener@238b521e, org.springframework.boot.test.autoconfigure.web.servlet.MockMvcPrintOnlyOnFailureTestExecutionListener@1b39fd82, org.springframework.boot.test.autoconfigure.web.servlet.WebDriverTestExecutionListener@3e2fc448, org.springframework.boot.test.autoconfigure.webservices.client.MockWebServiceServerTestExecutionListener@21680803, org.springframework.boot.test.mock.mockito.ResetMocksTestExecutionListener@588ab592]
00:55:51.264 [Test worker] DEBUG org.springframework.test.context.support.AbstractDirtiesContextTestExecutionListener - Before test class: context [DefaultTestContext@6d1310f6 testClass = WebsocketChatApplicationTests, testInstance = [null], testMethod = [null], testException = [null], mergedContextConfiguration = [WebMergedContextConfiguration@3228d990 testClass = WebsocketChatApplicationTests, locations = '{}', classes = '{class com.dlrtn.websocket.chat.WebsocketChatApplication}', contextInitializerClasses = '[]', activeProfiles = '{}', propertySourceLocations = '{}', propertySourceProperties = '{org.springframework.boot.test.context.SpringBootTestContextBootstrapper=true}', contextCustomizers = set[org.springframework.boot.test.autoconfigure.actuate.metrics.MetricsExportContextCustomizerFactory$DisableMetricExportContextCustomizer@25b2cfcb, org.springframework.boot.test.autoconfigure.properties.PropertyMappingContextCustomizer@0, org.springframework.boot.test.autoconfigure.web.servlet.WebDriverContextCustomizerFactory$Customizer@4504d271, org.springframework.boot.test.context.filter.ExcludeFilterContextCustomizer@2b5825fa, org.springframework.boot.test.json.DuplicateJsonObjectContextCustomizerFactory$DuplicateJsonObjectContextCustomizer@163d04ff, org.springframework.boot.test.mock.mockito.MockitoContextCustomizer@0, org.springframework.boot.test.web.client.TestRestTemplateContextCustomizer@79145d5a, org.springframework.boot.test.context.SpringBootTestArgs@1, org.springframework.boot.test.context.SpringBootTestWebEnvironment@1b410b60], resourceBasePath = 'src/main/webapp', contextLoader = 'org.springframework.boot.test.context.SpringBootContextLoader', parent = [null]], attributes = map['org.springframework.test.context.web.ServletTestExecutionListener.activateListener' -> true]], class annotated with @DirtiesContext [false] with mode [null].

**Comment by dlrtn** [comment]
버그는 직접 찾을건데.. 15시까지 못찾으면.. 힌트 주십쇼.. 핸임..

File: build.gradle [hunk 1]
```diff
@@ -29,13 +36,27 @@ dependencies {
...
+    implementation "org.springframework.boot:spring-boot-starter-security"
+    implementation "org.springframework.boot:spring-boot-starter-jdbc"
+
+    // thymeleaf
+    implementation "org.thymeleaf.extras:thymeleaf-extras-springsecurity5"
+
+    // mybatis
+    implementation "org.mybatis.spring.boot:mybatis-spring-boot-starter:$mybatis_spring_boot_starter_version"
+    implementation "org.mybatis:mybatis-typehandlers-jsr310:$mybatis_typehandlers_jsr310_version"
+    implementation "org.mybatis:mybatis-spring:$mybatis_spring_version"
```
Comment (woojae9488) [inline build.gradle]: 두 개의 의존성은 mybatis-spring-boot-starter 안에 포함되어 있을 거 같은데 별도로 필요한 부분일까요?

File: build.gradle [hunk 2]
```diff
@@ -29,13 +36,27 @@ dependencies {
     implementation "org.springframework.boot:spring-boot-starter-web"
     implementation "org.springframework.boot:spring-boot-starter-websocket"
     implementation "org.springframework.boot:spring-boot-starter-validation"
+    implementation "org.springframework.boot:spring-boot-starter-security"
+    implementation "org.springframework.boot:spring-boot-starter-jdbc"
+
+    // thymeleaf
+    implementation "org.thymeleaf.extras:thymeleaf-extras-springsecurity5"
```
Comment (woojae9488) [inline build.gradle:45]: spring boot starter thymeleaf를 사용하는게 좋을 거 같습니다.
(https://mvnrepository.com/artifact/org.springframework.boot/spring-boot-starter-thymeleaf
implementation 'org.springframework.boot:spring-boot-starter-thymeleaf:2.6.5')

File: build.gradle [hunk 3]
```diff
@@ -19,6 +19,13 @@ repositories {
 }
 
 ext {
+    mariadb_java_client_version = "2.3.0"
+    mybatis_spring_boot_starter_version = "2.2.0"
+    mybatis_typehandlers_jsr310_version = "1.0.2"
+    mybatis_spring_version = "2.0.6"
```
Comment (woojae9488) [inline build.gradle]: 아래 의존성 순서에 따라 변수도 정렬해주세요~!

File: build.gradle [hunk 4]
```diff
@@ -29,13 +36,27 @@ dependencies {
...
+    // thymeleaf
+    implementation "org.thymeleaf.extras:thymeleaf-extras-springsecurity5"
+
+    // mybatis
+    implementation "org.mybatis.spring.boot:mybatis-spring-boot-starter:$mybatis_spring_boot_starter_version"
+    implementation "org.mybatis:mybatis-typehandlers-jsr310:$mybatis_typehandlers_jsr310_version"
+    implementation "org.mybatis:mybatis-spring:$mybatis_spring_version"
+
+    // mariadb
+    implementation "org.mariadb.jdbc:mariadb-java-client:$mariadb_java_client_version"
```
Comment (woojae9488) [inline build.gradle]: 이건 어디 필요한 걸까요?
spring-boot-starter-jdbc에 의존성이 있는 건 아닐까요?

File: src/main/java/com/dlrtn/websocket/chat/WebsocketChatApplication.java [hunk 5]
```diff
@@ -9,5 +9,4 @@
 	public static void main(String[] args) {
 		SpringApplication.run(WebsocketChatApplication.class, args);
 	}
-
-}
+}
```
Comment (woojae9488) [inline src/main/java/com/dlrtn/websocket/chat/WebsocketChatApplication.java:13]: 마지막 줄에는 개행을 넣는 게 좋을 거 같습니다~!

Comment (woojae9488) [reply src/main/java/com/dlrtn/websocket/chat/WebsocketChatApplication.java:13]: (아래 클래스 파일들도 동일)

File: src/main/java/com/dlrtn/websocket/chat/service/UserService.java [hunk 6]
```diff
@@ -0,0 +1,36 @@
...
+import java.text.SimpleDateFormat;
+import java.util.Date;
+
+@Service
+@RequiredArgsConstructor
+public class UserService {
+    // 회원가입 시 저장시간을 넣어줄 DateTime 형
+    SimpleDateFormat format = new SimpleDateFormat ( "yyyy-MM-dd HH:mm:sss");
+    Date time = new Date();
+    String localTime = format.format(time);
```
Comment (woojae9488) [inline src/main/java/com/dlrtn/websocket/chat/service/UserService.java]: SimpleDateFormat은 multithread 환경에서 thread-safe하지 않은 것으로 알고 있어서 빈 내부에서 공통 속성으로 사용하기에는 좋지 않은 거 같습니다.
사용하시려면 DateTimeFormater를 사용하시고, Date 또한 Java에서 문제가 많은 클래스로 현재는 LocalDate, LocalDateTime을 사용하는 것이 바람직합니다.
(https://jeong-pro.tistory.com/163)

참고로 빈 내부에서 공통 속성을 가지고 있는 것은 멀티스레드 환경에서 race-condition 또는 성능 저하(락이 걸린 경우)를 일으킬 가능성이 높습니다.
되도록이면 메소드 안에서 처리하는 것이 좋습니다.

File: src/main/java/com/dlrtn/websocket/chat/service/UserService.java [hunk 7]
```diff
@@ -0,0 +1,36 @@
...
+
+    @Autowired
+    UserMapper userMapper;
```
Comment (woojae9488) [inline src/main/java/com/dlrtn/websocket/chat/service/UserService.java]: autowired 보다는 클래스에 @RequiredArgsAnnotation을 이용해서 생성자 주입으로 넣어주는게 더 선호되는 거 같습니다.

File: src/main/java/com/dlrtn/websocket/chat/service/UserService.java [hunk 8]
```diff
@@ -0,0 +1,36 @@
...
+
+    @Transactional
+    public void joinUser(UserVo userVo){
+        BCryptPasswordEncoder passwordEncoder = new BCryptPasswordEncoder();
+        userVo.setUserPw(passwordEncoder.encode(userVo.getUserPw()));
+        userVo.setUserAuth("USER");
+        userVo.setAppendDate(localTime);
+        userVo.setUpdateDate(localTime);
```
Comment (woojae9488) [inline src/main/java/com/dlrtn/websocket/chat/service/UserService.java]: Vo 객체와 Dto 객체에 대해 한 번 공부해보는 것도 좋을 거 같습니다.
그리고 setter는 객체의 불변성을 없앨 가능성이 높기 때문에 객체의 불변성을 유지할 수 있는 Builder 패턴을 사용하는 것이 좋습니다.

File: src/main/java/com/dlrtn/websocket/chat/service/UserService.java [hunk 9]
```diff
@@ -0,0 +1,36 @@
...
+    SimpleDateFormat format = new SimpleDateFormat ( "yyyy-MM-dd HH:mm:sss");
+    Date time = new Date();
+    String localTime = format.format(time);
+
+    @Autowired
+    UserMapper userMapper;
+
+    @Transactional
+    public void joinUser(UserVo userVo){
+        BCryptPasswordEncoder passwordEncoder = new BCryptPasswordEncoder();
```
Comment (woojae9488) [inline src/main/java/com/dlrtn/websocket/chat/service/UserService.java:30]: BCryptPasswordEncoder는 thread-safe 할까요?
메소드 내부에서 객체 생성 시 생성 배용은 무시해도 될 정도일까요?

File: src/main/resources/query/UserMapper.xml [hunk 10]
```diff
@@ -0,0 +1,11 @@
...
+<!DOCTYPE mapper
+        PUBLIC "-//mybatis.org//DTD Mapper 3.0//EN"
+        "http://mybatis.org/dtd/mybatis-3-mapper.dtd">
+<mapper namespace="com.dlrtn.websocket.chat.mapper.UserMapper">
+    <insert id="saveUser" parameterType="com.dlrtn.websocket.chat.vo.UserVo">
+        INSERT INTO TB_USER
+        (USER_ID, USER_PW, USER_NAME, USER_AUTH, APPEND_DATE, UPDATE_DATE)
+        VALUES(#{userId},#{userPw},#{userName},#{userAuth},#{appendDate},#{updateDate});
+    </insert>
+</mapper>
```
Comment (woojae9488) [inline src/main/resources/query/UserMapper.xml]: 쿼리에 대한 포맷도 한 번 공부해보는게 좋을 거 같습니다.
git 로그들을 알아보기도 쉽고 콤마(,) 이후에는 개행을 넣는 것이 좀 더 가독성이 높아서요~!

File: src/main/resources/templates/signup.html [hunk 11]
```diff
@@ -0,0 +1,30 @@
...
+    </div>
+    <div class="form-group">
+      <label for="inputPassword4">password</label>
+      <input type="password" class="form-control" id="inputPassword4" name="userPw" placeholder="사용자 비밀번호">
+    </div>
+    <button type="submit" class="btn btn-primary">가입 완료</button>
+  </div>
+</form>
+</body>
+</html>
```
Comment (woojae9488) [inline src/main/resources/templates/signup.html:30]: 프론트 로직에 대한 리뷰는 따로 안 달도록 하겠습니다~!(제가 달아봤자 의미없을 거 같아서요!)

**Review by woojae9488 — CHANGES_REQUESTED** [review]

**Comment by woojae9488** [comment]
> 버그는 직접 찾을건데.. 15시까지 못찾으면.. 힌트 주십쇼.. 핸임..

@dlrtn 
로그 확인해보면 `WebsocketChatApplicationTests` 여기에 이슈가 좀 있는 거 같은데
제가 실행했을 때는 잘 동작하네요~!

**Pull request closed without merging**